Assets/MRTK/Tools/RuntimeTools/Tools/MixedRealityInteractionTool/MixedRealityInteractionTool.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Tools.Runtime
+{
+    /// <summary>
+    /// Displays active left and right handed controllers with available interactions and their current state.
+    /// </summary>
+    [AddComponentMenu("Scripts/MRTK/Tools/MixedRealityInteractionTool")]
+    public class MixedRealityInteractionTool : MonoBehaviour
+    {
+        [SerializeField]
+        [Tooltip("Used for displaying all detected input source names.")]
+        private TextMesh listControllersTextMesh = null;
+
+        [SerializeField]
+        [Tooltip("Used for displaying the left handed input source interactions.")]
+        private TextMesh leftHandedControllerTextMesh = null;
+
+        [SerializeField]
+        [Tooltip("Used for displaying the right handed input source interactions.")]
+        private TextMesh rightHandedControllerTextMesh = null;
+
+        private void Update()
+        {
+            if (listControllersTextMesh == null || leftHandedControllerTextMesh == null || rightHandedControllerTextMesh == null)
+            {
+                return;
+            }
+
+            listControllersTextMesh.text = string.Empty;
+            leftHandedControllerTextMesh.text = string.Empty;
+            rightHandedControllerTextMesh.text = string.Empty;
+
+            HashSet<IMixedRealityController> controllers = CoreServices.InputSystem?.DetectedControllers;
+
+            listControllersTextMesh.text = $"Detected {controllers?.Count} input source{(controllers?.Count > 1 ? "s:" : controllers?.Count != 0 ? ":" : "s")}\n";
+
+            if (controllers == null)
+            {
+                return;
+            }
+
+            foreach (IMixedRealityController controller in controllers)
+            {
+                listControllersTextMesh.text += $"{controller.InputSource.SourceName}\n";
+
+                TextMesh textMesh;
+
+                if (controller.ControllerHandedness == Handedness.Left)
+                {
+                    textMesh = leftHandedControllerTextMesh;
+                }
+                else if (controller.ControllerHandedness == Handedness.Right)
+                {
+                    textMesh = rightHandedControllerTextMesh;
+                }
+                else
+                {
+                    continue;
+                }
+
+                textMesh.text = $"{controller.InputSource.SourceName}\n\n";
+
+                foreach (MixedRealityInteractionMapping interaction in controller.Interactions)
```

Comment:
Is it possible to order these interactions by groups (bool, float, vector, quaternion) and then make sure to display them in order to make things a bit easier to read?

Reply:
I decided to keep this as-is for now, since the interactions as they're defined in code are often grouped by button (for example, touchpad touch, touchpad press, and touchpad axes all next to each other), which i've found useful for testing a single button and being able to look at a few lines next to each other